### PR TITLE
[AAP-59525] Implement APIVersionRegistry, Filesystem‑Based Version Discovery, DynamicClassLoader, API Version Detection, Single Connection Plugin, Unit TC CI

### PR DESCRIPTION
**Pull request by nickbhasin**
## Description

- What is being changed?
Implemented dynamic version negotiation in APIVersionRegistry, DynamicClassLoader, API Version Detection, Single Connection Plugin, Unit TC CI.
- Why is this change needed?
To enable fallback mechanism and version detection, unit tc enablement.
- How does this change address the issue?
The system now pings the gateway, identifies the version (v1/v2), and successfully falls back to the highest local implementation (v2) if an unsupported version (v3) is detected.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [x] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

File: plugins/plugin_utils/api/v2/user.py [hunk 1]
```diff
@@ -178,26 +178,6 @@ def get_endpoint_operations(cls) -> Dict[str, EndpointOperation]:
     def get_lookup_field(cls) -> str:
         return "username"
 
-    def to_api(self, context: Union[TransformContext, Dict[str, Any]]) -> "APIUser_v2":
```
Comment (rohitthakur2590) [inline plugins/plugin_utils/api/v2/user.py]: why we have removed this

File: plugins/plugin_utils/api/v1/user.py [hunk 2]
```diff
@@ -223,51 +223,6 @@ def get_lookup_field(cls) -> str:
         """
         return 'username'
 
-    def to_api(self, context: Union[TransformContext, Dict[str, Any]]) -> 'APIUser_v1':
```
Comment (rohitthakur2590) [inline plugins/plugin_utils/api/v1/user.py]: any reason to remove this one

File: plugins/plugin_utils/manager/platform_manager.py [hunk 3]
```diff
@@ -116,23 +116,11 @@ def __init__(self, config: GatewayConfig):
...
-            logger.info(f"PlatformService: API version detected: v{self.api_version}")
-        except Exception as e:
-            logger.warning(f"PlatformService: Version detection failed: {e}, defaulting to v1")
-            self.api_version = '1'  # CRITICAL: Always default to '1' on failure
-        
-        # Final validation - ensure api_version is '1' (AAP Gateway currently only supports v1)
-        if self.api_version != '1':
-            logger.warning(f"PlatformService: Detected version '{self.api_version}' but AAP Gateway only supports v1, forcing to '1'")
-            self.api_version = '1'
+        # Detect API version dynamically
```
Comment (rohitthakur2590) [inline plugins/plugin_utils/manager/platform_manager.py:117]: what happens if detected version is not supported

File: tests/unit/modules/test_registry.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import unittest
+
+from ansible_collections.ansible.platform.plugins.plugin_utils.platform.registry import APIVersionRegistry
+from ansible_collections.ansible.platform.plugins.plugin_utils.platform.loader import DynamicClassLoader
+
+class TestUserModule(unittest.TestCase):
```
Comment (rohitthakur2590) [inline tests/unit/modules/test_registry.py]: we need to conver all the scenarios.

File: plugins/plugin_utils/manager/platform_manager.py (hunk 3 above)
Comment (nickbhasin) [reply plugins/plugin_utils/manager/platform_manager.py:117]: PlatformService falls back to the highest locally supported version.

In platform_manager.py, the _detect_api_version() method checks the server's version against registry.get_supported_versions(). If it's not a match, it defaults to registry.get_latest_version() so the collection can still attempt backwards compatible operations rather than crashing.

File: plugins/plugin_utils/api/v2/user.py (hunk 1 above)
Comment (nickbhasin) [reply plugins/plugin_utils/api/v2/user.py]: If we kept to_api, we would have to manually write out and maintain a dictionary for every single resource like {"username": self.username, "email": self.email, ...} etc. Every time a new API field was added in the future, we will have to update that manual dictionary mapping.

By using python dataclasses, the platform manager simply calls python's built-in dataclasses.asdict() to automatically convert the entire object into a dictionary for the API payload. This completely removes the need to manually map fields one-by-one, keeping the code much cleaner and preventing human error.

File: tests/unit/modules/test_registry.py (hunk 4 above)
Comment (nickbhasin) [reply tests/unit/modules/test_registry.py]: added all the scenarios.

**Review by rohitthakur2590 — COMMENTED** [review]
Overall this looks good, but I don't see any unit test running or logs .Could you point me to the  unit test run

**Comment by djdanielsson** [comment]
once you fix linting I will approve the integration tests to verify those pass

**Comment by nickbhasin** [comment]
Changes of this PR are included in previous PRs by @rohitthakur2590 and also present in this PR - https://github.com/ansible/ansible.platform/pull/147.
